docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/rs-configuration.md

```diff
@@ -182,16 +182,259 @@ The capacity usage on a pool can be monitored using [exporter metrics](../replic
-2. **volumeCommitment** parameter specifies the minimum amount of free space that must be present in each replica pool in order to create new replicas for an existing volume. This value is specified as a percentage of the volume size.
-3. **volumeCommitmentInitial** minimum amount of free space that must be present in each replica pool in order to create new replicas for a new volume. This value is specified as a percentage of the volume size.
+2. **volumeCommitment** parameter specifies the minimum amount of free space that must be present in each replica pool to create new replicas for an existing volume. This value is specified as a percentage of the volume size.
+3. **volumeCommitmentInitial** minimum amount of free space that must be present in each replica pool to create new replicas for a new volume. This value is specified as a percentage of the volume size.
 
 
 :::note
 - By default, the volumes are provisioned as `thick`. 
 - For a pool of a particular size, say 10 Gigabytes, a volume > 10 Gigabytes cannot be created, as Replicated PV Mayastor currently does not support pool expansion.
-- The replicas for a given volume can be either all thick or all thin. Same volume cannot have a combination of thick and thin replicas.
+- The replicas for a given volume can be either all thick or all thin. The same volume cannot have a combination of thick and thin replicas.
 :::
 
+### "nodeAffinityTopologyLabel"
+
+The parameter 'nodeAffinityTopologyLabel' will allow the placement of replicas on the node that exactly matches the labels defined in the storage class.
+For the case shown below, the replica of the volume will be placed on `worker-node-1` and `worker-node-3` only as they match the labels specified under `noolAffinityTopologyLabel` in storage class which is equal to zone=us-west-1.
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  nodeAffinityTopologyLabel: |
+    zone: us-west-1
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels to the nodes using the below command:
+
+**Command**
+```text
+kubectl mayastor label node worker-node-1 zone=us-west-1
+kubectl mayastor label node worker-node-2 zone=eu-east-1
+kubectl mayastor label node worker-node-3 zone=us-west-1
+ ```
+
+**Command (Get nodes)**
+ ```text
+kubectl mayastor get nodes -n openebs --show-labels
+ID             GRPC ENDPOINT        STATUS  LABELS
+worker-node-1  65.108.91.181:10124  Online  zone=eu-west-1
+worker-node-3  65.21.4.103:10124    Online  zone=eu-east-1
+worker-node-3  37.27.13.10:10124    Online  zone=us-west-1
+```
+
+### "NodeHasTopologyKey"
+
+The parameter 'NodeHasTopologyKey' will allow the placement of replicas on the node that has label keys that are identical to the keys specified in the storage class.
```

Comment:
label keys identical to the keys...

Reply:
@sinhaashish do we support matching of multiple keys?

Reply:
yes that the reason we have `|` in storage class.